lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -69,6 +69,79 @@ public void setScorer(Scorable scorer) throws IOException {
     }
   }
 
+  private abstract class TopFieldLeafCollector extends MultiComparatorLeafCollector {
```

Comment:
Should we maybe pull this out into its own `TopFieldLeafCollector.java` source?

Reply:
Well it has many reliances on `TopFieldCollector`'s members, and is never created on its own, so I think it's probably best as an inner class.

Reply:
I was mostly focused on the code duplicattion. However there are other opportunities to simplify - eg we could merge with `MultiComparatorLeafCollector`.

Reply:
OK let's do that in followon issue.  Progress not perfection!